app/controllers/dashboard_controller.rb

```diff
@@ -10,21 +10,44 @@ def sheets
   end
 
   def save
-    deletes = Array.wrap(params['delete']).concat(params['add'].keys)
+    return head :bad_request if invalid_day_periods?
+
+    deletes = Array.wrap(punch_params['delete']).concat(params_add.keys)
```

Comment:
What do you think about moving this entire logic for this `create` method into a service object? Something like `CreatePunchesInBatch.call(additions: [], deletions: [])`

Reply:
Yeah, would be great!